go/parquet/metadata/column_chunk.go

```diff
@@ -0,0 +1,385 @@
+	ok, err := c.StatsSet()
+	if err != nil {
+		return nil, err
+	}
+
+	if ok {
+		return c.possibleStats, nil
+	}
+	return nil, nil
+}
+
+// ColumnChunkMetaDataBuilder is used during writing to construct metadata
+// for a given column chunk while writing, providing a proxy around constructing
+// the actual thrift object.
+type ColumnChunkMetaDataBuilder struct {
+	chunk  *format.ColumnChunk
+	props  *parquet.WriterProperties
+	column *schema.Column
+
+	compressedSize int64
+}
+
+func NewColumnChunkMetaDataBuilder(props *parquet.WriterProperties, column *schema.Column) *ColumnChunkMetaDataBuilder {
+	return NewColumnChunkMetaDataBuilderWithContents(props, column, format.NewColumnChunk())
+}
+
+// NewColumnChunkMetaDataBuilderWithContents will construct a builder and start it with the provided
+// column chunk information rather than with an empty column chunk.
+func NewColumnChunkMetaDataBuilderWithContents(props *parquet.WriterProperties, column *schema.Column, chunk *format.ColumnChunk) *ColumnChunkMetaDataBuilder {
+	b := &ColumnChunkMetaDataBuilder{
+		props:  props,
+		column: column,
+		chunk:  chunk,
+	}
+	b.init(chunk)
+	return b
+}
+
+// Contents returns the underlying thrift ColumnChunk object so that it can be used
+// for constructing or duplicating column metadata
+func (c *ColumnChunkMetaDataBuilder) Contents() *format.ColumnChunk { return c.chunk }
+
+func (c *ColumnChunkMetaDataBuilder) init(chunk *format.ColumnChunk) {
+	c.chunk = chunk
+	if !c.chunk.IsSetMetaData() {
+		c.chunk.MetaData = format.NewColumnMetaData()
+	}
+	c.chunk.MetaData.Type = format.Type(c.column.PhysicalType())
+	c.chunk.MetaData.PathInSchema = schema.ColumnPathFromNode(c.column.SchemaNode())
+	c.chunk.MetaData.Codec = format.CompressionCodec(c.props.CompressionFor(c.column.Path()))
+}
+
+func (c *ColumnChunkMetaDataBuilder) SetFilePath(val string) {
+	c.chunk.FilePath = &val
+}
+
+// Descr returns the associated column descriptor for this column chunk
+func (c *ColumnChunkMetaDataBuilder) Descr() *schema.Column { return c.column }
+
+func (c *ColumnChunkMetaDataBuilder) TotalCompressedSize() int64 {
+	// if this column is encrypted, after Finish is called, the MetaData
+	// field is set to nil and we store the compressed size so return that
+	if c.chunk.MetaData == nil {
+		return c.compressedSize
+	}
+	return c.chunk.MetaData.GetTotalCompressedSize()
+}
+
+func (c *ColumnChunkMetaDataBuilder) SetStats(val EncodedStatistics) {
+	c.chunk.MetaData.Statistics = val.ToThrift()
+}
+
+// Finish finalizes the metadata with the given offsets,
+// flushes any compression that needs to be done, and performs
+// any encryption if an encryptor is provided.
+func (c *ColumnChunkMetaDataBuilder) Finish(nvalues, dictPageOffset, indexPageOffset, dataPageOffset, compressed, uncompressed int64, hasDict, dictFallback bool, dictEncodingStats, dataEncodingStats map[parquet.Encoding]int32, metaEncryptor encryption.Encryptor) error {
```

Comment:
does it make sense to make a structure here to pass through instead of all these parameters?

Reply:
added two struct types to better organize the params and shrink the number of them